terraform/main.tf

```diff
@@ -0,0 +1,194 @@
+
+  }
+}
+
+data "archive_file" "lambda_log_connector" {
+  type = "zip"
+
+  source_dir  = local.lambda_src_path
+  output_path = "${path.module}/.tmp/${random_uuid.lambda_src_hash.result}.zip"
+  depends_on = [null_resource.install_dependencies]
+}
+
+resource "aws_s3_object" "lambda_log_connector" {
+  bucket = aws_s3_bucket.log_connector_lambda_bucket.id
+  key    = "lambda.zip"
+  source = data.archive_file.lambda_log_connector.output_path
+  etag = filemd5(data.archive_file.lambda_log_connector.output_path)
+}
+
+resource "aws_lambda_function" "log_connector" {
+  function_name = "SiemLogConnector"
+  description   = "Node serverless function to push AWS Logs to SecOps Endpoint"
+
+  s3_bucket = aws_s3_bucket.log_connector_lambda_bucket.id
+  s3_key    = aws_s3_object.lambda_log_connector.key
+
+  runtime = "nodejs14.x"
+  handler = "./index.handler"
+
+  environment {
+    variables = {
+      "LOG_CONSUMER_ENDPOINT" = var.log_consumer_endpoint
+    }
+  }
+
+  source_code_hash = data.archive_file.lambda_log_connector.output_base64sha256
+  role             = aws_iam_role.lambda_exec.arn
+}
+
+
+resource "aws_iam_role" "lambda_exec" {
+  name = "SecOps_log_transfer_role"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_iam_policy" "lambda_policy" {
+  name        = "SecOps_log_transfer_policy"
+  description = "Policy for the Secops Connection Lambda"
+
+  policy = <<EOF
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "VisualEditor0",
+            "Effect": "Allow",
+            "Action": [
+                "s3:GetObjectAcl",
+                "s3:GetObject",
+                "kms:Decrypt",
+                "kms:GenerateDataKey",
+                "kms:DescribeKey"
+            ],
+            "Resource": [
+                "arn:aws:s3:::*/*",
+                "arn:aws:kms:*:${data.aws_caller_identity.current.account_id}:key/*"
+            ]
+        },
+        {
+            "Sid": "VisualEditor1",
+            "Effect": "Allow",
+            "Action": [
+                "ssm:GetParameters",
+                "ssm:GetParameter"
+            ],
+            "Resource": "arn:aws:ssm:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:parameter/secops/siem/cert"
+        },
+         {
+          "Sid": "VisualEditor2",
+          "Effect": "Allow",
+          "Action": [
+              "logs:CreateLogStream",
+              "logs:CreateLogGroup",
+              "logs:PutLogEvents"
+          ],
+          "Resource": "*"
+        }
+    ]
+}
+EOF
+}
+
+resource "aws_iam_role_policy_attachment" "test-attach" {
+  role       = aws_iam_role.lambda_exec.name
+  policy_arn = aws_iam_policy.lambda_policy.arn
+}
+
+
+data "aws_s3_bucket" "trigger" {
```

Comment:
You could use `for_each` on the var.trigger_buckets variable if it is refactored to use a list or map. This will simplify the code.